middleware/opencensus_tracing.go

```diff
@@ -75,3 +97,72 @@ func getSpanContext(r *http.Request) (sc trace.SpanContext, ok bool) {
 
 	return propagation.FromBinary(bin)
 }
+
+func closeSpan(span *trace.Span, w *responseWriterDecorator) {
+	if w.StatusCode() < 400 {
+		span.SetStatus(trace.Status{
+			Code:    trace.StatusCodeOK,
+			Message: "OK",
+		})
+	} else {
+		span.SetStatus(trace.Status{
+			Code:    trace.StatusCodeUnknown,
+			Message: fmt.Sprintf("Response status code: %d", w.StatusCode()),
+		})
+	}
+	span.End()
+}
+
+func addSpanMessageReceiveEvent(span *trace.Span, r *http.Request) {
+	eIDString := r.Header.Get(headerNameOpencensusSpanEventIDKey)
+	eID, _ := strconv.ParseInt(eIDString, 10, 64)
+
+	span.AddMessageReceiveEvent(eID, r.ContentLength, 0)
+}
+
+func addSpanMessageSentEvent(span *trace.Span, r *http.Request) {
+	eID := generateEventID()
+	eIDString := strconv.FormatInt(eID, 10)
+	r.Header.Set(headerNameOpencensusSpanEventIDKey, eIDString)
+
+	span.AddMessageSendEvent(eID, r.ContentLength, 0)
+}
+
+func setSpanRequestPayloadAttribute(span *trace.Span, body *requestBodyDecorator) {
+	payload := string(body.Payload())
+	if len(payload) > payloadSizeLimit {
+		payload = payload[:payloadSizeLimit]
```

Comment:
maybe instead of truncate, we want to add some placeholder, like `(payload bigger than)`

Reply:
https://github.com/krzysztofreczek/chi-opencensus-tracing/pull/1/commits/944c4fb23efc7f42ac1923b6ded1364bab34e21a